lib/pedicel/base.rb

```diff
@@ -108,24 +112,35 @@ def verify_signature(ca_certificate_pem: Pedicel.config[:apple_root_ca_g3_cert_p
         raise SignatureError, "invalid PKCS #7 signature: #{e.message}"
       end
 
-      # 1.a
-      # Ensure that the certificates contain the correct custom OIDs: (...).
-      # The value for these marker OIDs doesn't matter, only their presence.
-      leaf, intermediate = self.class.verify_signature_certificate_oids(signature: s)
-
       begin
-        root = OpenSSL::X509::Certificate.new(ca_certificate_pem)
+        trusted_root = OpenSSL::X509::Certificate.new(ca_certificate_pem)
       rescue => e
-        raise CertificateError, "invalid root certificate: #{e.message}"
+        raise CertificateError, "invalid trusted root certificate: #{e.message}"
       end
 
+      # 1.a
+      # Ensure that the certificates contain the correct custom OIDs: (...).
+      # The value for these marker OIDs doesn't matter, only their presence.
+      leaf, intermediate, root = self.class.extract_certificates(signature: s)
+      # Implicit since these are the ones extracted.
+      # The only certificate that will be allowed to be extracted other than the
+      # leaf and intermediate is the root.
```

Comment:
`Base.extract_certificates` just extracts `leaf`, `intermediate` and `other`. However, due to `Base::verify_root_certificate` we verify that this "`other`" certificate must be the root, the variable name is `root`. Perhaps L124 should just read

```ruby
leaf, intermediate, other = self.class.extract_certificates(signature: s)
```

And L132 should then read

```ruby
self.class.verify_root_certificate(trusted_root: trusted_root, root: other)
```

WDYT?

Reply:
```
diff --git a/lib/pedicel/base.rb b/lib/pedicel/base.rb
index 2e1bb59..d08a9e2 100644
--- a/lib/pedicel/base.rb
+++ b/lib/pedicel/base.rb
@@ -121,15 +121,14 @@ module Pedicel
       # 1.a
       # Ensure that the certificates contain the correct custom OIDs: (...).
       # The value for these marker OIDs doesn't matter, only their presence.
-      leaf, intermediate, root = self.class.extract_certificates(signature: s)
+      leaf, intermediate, other = self.class.extract_certificates(signature: s)
       # Implicit since these are the ones extracted.
-      # The only certificate that will be allowed to be extracted other than the
-      # leaf and intermediate is the root.

       # 1.b
       # Ensure that the root CA is the Apple Root CA - G3. (...)
       if root
-        self.class.verify_root_certificate(trusted_root: trusted_root, root: root)
+        self.class.verify_root_certificate(trusted_root: trusted_root, root: other)
+        # Allow no other certificates than the root.
       #else
         # root is not extracted from the signature, and thus, we trust the
         # trusted root.
```

Reply:
Diff updated multiple times — don't trust the mailed version 🙂

Reply:
Now it makes perfect sense to me :+1: :slightly_smiling_face: 

I somehow interpreted "that will be allowed to be extracted" as in some method will only allow us to extract those three certificates, when the reality is that we just want to make sure that the third is indeed the root.